detekt-rules/src/main/kotlin/OrtImportOrder.kt

```diff
@@ -38,24 +40,93 @@ class OrtImportOrder : Rule() {
-            val finding = CodeSmell(
-                issue,
-                // Use the message as the name to also see it in CLI output and not only in the report files.
-                Entity.from(importList).copy(name = message),
-                message
-            )
-            report(finding)
+        val children = importList.node.getChildren(null)
+        if (children.isNotEmpty()) {
+            val importPaths = children.mapNotNull {
+                when (it.psi) {
+                    is KtImportDirective -> (it.psi as KtImportDirective).importPath.toString()
+                    is PsiWhiteSpace -> if (it.textLength > 1) "" else null
+                    else -> null
+                }
+            }
+
+            val importPathsWithoutBlankLines = importPaths.filter(String::isNotEmpty).toMutableList()
+
+            val expectedImportOrder = createExpectedImportOrder(importPathsWithoutBlankLines)
+
+            if (importPaths != expectedImportOrder) {
+                val path = importList.containingKtFile.absolutePath()
+                val message = "Imports in file '$path' are not sorted alphabetically"
+                val finding = CodeSmell(
+                    issue,
+                    // Use the message as the name to also see it in CLI output and not only in the report files.
+                    Entity.from(importList).copy(name = message),
+                    message
+                )
+                report(finding)
+            }
+        }
+    }
+
+    private fun createExpectedImportOrder(importPaths: List<String>): List<String> {
+        val importPathsWithDot = mutableListOf<String>()
+        val importPathsWithoutDot = mutableListOf<String>()
+        val expectedImportPaths = mutableListOf<String>()
+
+        importPaths.forEach {
+            if (it.indexOf(".") == -1) {
+                importPathsWithoutDot.add(it)
+            } else {
+                importPathsWithDot.add(it)
+            }
+        }
+
+        importPathsWithoutDot.sort()
+        importPathsWithDot.sort()
+        val sortedImportPathsWithDotAndBlankLines = createImportListWithBlankLines(importPathsWithDot)
+
+        if (importPathsWithoutDot.isNotEmpty()) {
+            expectedImportPaths.addAll(importPathsWithoutDot)
+            expectedImportPaths.add("")
+        }
+
+        expectedImportPaths.addAll(sortedImportPathsWithDotAndBlankLines)
+        return expectedImportPaths
+    }
+
+    private fun createImportListWithBlankLines(importPaths: List<String>): List<String> {
+        val pathsWithBlankLines = mutableListOf<String>()
+
+        var previousTopLevel = getTopLevelPackage(importPaths[0])
+        importPaths.forEach {
+            val currentTopLevel = getTopLevelPackage(it)
+            if (currentTopLevel != previousTopLevel) {
+                pathsWithBlankLines.add("")
+            }
+
+            pathsWithBlankLines.add(it)
+            previousTopLevel = currentTopLevel
+        }
```

Comment:
I'm not sure whether it's really much nicer (as I dislike the final `removeLast()`), but this could be written as

        importPaths.groupBy { getTopLevelPackage(it) }.forEach {
            pathsWithBlankLines += it.value
            pathsWithBlankLines += ""
        }
        pathsWithBlankLines.removeLast()